### PR TITLE
E2E: Don't check for IP

### LINKED ISSUE #129: E2E tests are failing when deploying with globalnet
While working on #127 I discovered that globalnet isn't supported by the E2E tests.

The failures are recorded in the log - [log.txt](https://github.com/submariner-io/lighthouse/files/4608932/log.txt)

Excerpt:
```
2020-05-11T09:45:50.2149430Z [91m[1mâ€¢ Failure [136.424 seconds][0m
2020-05-11T09:45:50.2149647Z [dataplane] Test Service Discovery Across Clusters
2020-05-11T09:45:50.2154525Z [90m/go/src/github.com/submariner-io/lighthouse/test/e2e/dataplane/service_discovery.go:16[0m
2020-05-11T09:45:50.2154684Z   when a pod tries to resolve a service in a remote cluster
2020-05-11T09:45:50.2155033Z   [90m/go/src/github.com/submariner-io/lighthouse/test/e2e/dataplane/service_discovery.go:19[0m
2020-05-11T09:45:50.2155343Z     [91m[1mshould be able to discover the remote service successfully [It][0m
2020-05-11T09:45:50.2155633Z     [90m/go/src/github.com/submariner-io/lighthouse/test/e2e/dataplane/service_discovery.go:20[0m
2020-05-11T09:45:50.2155716Z 
2020-05-11T09:45:50.2156603Z     [91mFailed to verify if service IP is discoverable. expected execution result "; <<>> DiG 9.14.8 <<>> @100.90.0.10 nginx-demo.e2e-tests-dataplane-sd-pmx54.svc.cluster2.local +short\n; (1 server found)\n;; global options: +cmd\n;; connection timed out; no servers could be reached\n169.254.3.81" to contain "100.90.209.156"
2020-05-11T09:45:50.2156764Z     Unexpected error:
2020-05-11T09:45:50.2156844Z         <exec.CodeExitError>: {
2020-05-11T09:45:50.2156938Z             Err: {
2020-05-11T09:45:50.2157034Z                 s: "command terminated with exit code 9",
2020-05-11T09:45:50.2157133Z             },
2020-05-11T09:45:50.2157472Z             Code: 9,
2020-05-11T09:45:50.2157573Z         }
2020-05-11T09:45:50.2157872Z         command terminated with exit code 9
2020-05-11T09:45:50.2158333Z     occurred[0m

...

2020-05-11T09:47:46.3741029Z [91m[1mâ€¢ Failure [116.159 seconds][0m
2020-05-11T09:47:46.3742070Z [dataplane] Test Service Discovery Across Clusters
2020-05-11T09:47:46.3744907Z [90m/go/src/github.com/submariner-io/lighthouse/test/e2e/dataplane/service_discovery.go:16[0m
2020-05-11T09:47:46.3746916Z   when a pod tries to resolve a service which is present locally and in a remote cluster
2020-05-11T09:47:46.3750226Z   [90m/go/src/github.com/submariner-io/lighthouse/test/e2e/dataplane/service_discovery.go:25[0m
2020-05-11T09:47:46.3751495Z     [91m[1mshould resolve the local service [It][0m
2020-05-11T09:47:46.3755627Z     [90m/go/src/github.com/submariner-io/lighthouse/test/e2e/dataplane/service_discovery.go:26[0m
2020-05-11T09:47:46.3757270Z 
2020-05-11T09:47:46.3760909Z     [91mFailed to verify if service IP is discoverable
2020-05-11T09:47:46.3761064Z     Unexpected error:
2020-05-11T09:47:46.3761437Z         <exec.CodeExitError>: {
2020-05-11T09:47:46.3761536Z             Err: {
2020-05-11T09:47:46.3761666Z                 s: "command terminated with exit code 9",
2020-05-11T09:47:46.3761768Z             },
2020-05-11T09:47:46.3761844Z             Code: 9,
2020-05-11T09:47:46.3761940Z         }
2020-05-11T09:47:46.3762034Z         command terminated with exit code 9
2020-05-11T09:47:46.3762339Z     occurred[0m
```

### PR DESCRIPTION
**Pull request by vthapar**
Service Discovery E2E should only test if query to service name
resolves or not. It should not check for actual IP returned as that
will depend on deployment - globalnet enabled or not.

Fixes #129

### CODE REVIEW

File: test/e2e/dataplane/service_discovery.go [hunk 1]
```diff
@@ -84,7 +92,18 @@ func RunServiceDiscoveryLocalTest(f *framework.Framework) {
...
-	verifyClusterIpWithDig(f, framework.ClusterB, nginxServiceClusterC, netshootPodList, true)
+	cmd := []string{"curl", nginxServiceClusterC.Name}
+	stdout, _, err := f.ExecWithOptions(framework.ExecOptions{
+		Command:       cmd,
+		Namespace:     f.Namespace,
+		PodName:       netshootPodList.Items[0].Name,
+		ContainerName: netshootPodList.Items[0].Spec.Containers[0].Name,
+		CaptureStdout: true,
+		CaptureStderr: true,
+	}, framework.ClusterB)
```
Comment (mkolesnik) [inline test/e2e/dataplane/service_discovery.go:103]: Maybe put this duplicated monstrosity in a helper function?

It seems to be repeating quite a few times now with very minor variances that can be sent as function params..

Comment (vthapar) [reply test/e2e/dataplane/service_discovery.go:103]: > Maybe put this duplicated monstrosity in a helper function?
> 
> It seems to be repeating quite a few times now with very minor variances that can be sent as function params..

Yeah, doing that.

File: test/e2e/dataplane/service_discovery.go [hunk 2]
```diff
@@ -84,7 +92,18 @@ func RunServiceDiscoveryLocalTest(f *framework.Framework) {
...
+
+	Expect(err).NotTo(HaveOccurred())
+	Expect(stdout).To(ContainSubstring("Welcome to nginx!"))
```
Comment (sridhargaddam) [inline test/e2e/dataplane/service_discovery.go:106]: @vthapar as discussed over slack, please add a log which dumps the IP address returned by LH

File: test/e2e/dataplane/service_discovery.go [hunk 3]
```diff
@@ -119,13 +109,32 @@ func verifyClusterIpWithDig(f *framework.Framework, cluster framework.ClusterInd
...
 		if doesContain && !shouldContain {
-			return false, fmt.Sprintf("expected execution result %q not to contain %q", result, serviceIP), nil
+			return false || logOnly, fmt.Sprintf("expected execution result %q not to contain %q", result, serviceIP), nil
 		}
 
 		if !doesContain && shouldContain {
-			return false, fmt.Sprintf("expected execution result %q to contain %q", result, serviceIP), nil
+			return false || logOnly, fmt.Sprintf("expected execution result %q to contain %q", result, serviceIP), nil
 		}
+		By(fmt.Sprintf("Validating dig result: %q", result))
```
Comment (mkolesnik) [inline test/e2e/dataplane/service_discovery.go:118]: Shouldn't this be before the ifs?

File: test/e2e/dataplane/service_discovery.go [hunk 4]
```diff
@@ -119,13 +109,32 @@ func verifyClusterIpWithDig(f *framework.Framework, cluster framework.ClusterInd
...
 
 		return true, "", nil
 	})
 }
+
+func verifyCurl(f *framework.Framework, cluster framework.ClusterIndex, pod v1.Pod, target string, shouldPass bool) {
```
Comment (mkolesnik) [inline test/e2e/dataplane/service_discovery.go:124]:
```suggestion
func verifyCurl(f *framework.Framework, cluster framework.ClusterIndex, pod v1.Pod, target string, expectingSuccess bool) {
```

File: test/e2e/dataplane/service_discovery.go [hunk 5]
```diff
@@ -119,13 +109,32 @@ func verifyClusterIpWithDig(f *framework.Framework, cluster framework.ClusterInd
...
+	cmd := []string{"curl", target}
+	stdout, _, err := f.ExecWithOptions(framework.ExecOptions{
+		Command:       cmd,
+		Namespace:     f.Namespace,
+		PodName:       pod.Name,
+		ContainerName: pod.Spec.Containers[0].Name,
+		CaptureStdout: true,
+		CaptureStderr: true,
+	}, framework.ClusterB)
+	if shouldPass {
```
Comment (mkolesnik) [inline test/e2e/dataplane/service_discovery.go:134]:
```suggestion
	if expectingSuccess {
```

File: test/e2e/dataplane/service_discovery.go (hunk 3 above)
Comment (vthapar) [reply test/e2e/dataplane/service_discovery.go:118]: Oh right. I refactored my initial fix with logOnly, forgot to move it.

**Pull request merged**